src/dataDisplay/personPanel/personPanelDetails.jsx

```diff
@@ -891,28 +894,31 @@ function PersonPanelDetails(props) {
 
                     {children}
 
-                    {(!isEmpty(selectButtonProps) || !isEmpty(viewRecordButtonProps)) && (
-                        <div
-                            className={classes.actions}
-                        >
-                            {!isEmpty(selectButtonProps) && (
-                                <PersonPanelDetailsActionButton
-                                    label="Select"
-                                    // eslint-disable-next-line react/jsx-props-no-spreading
-                                    {...selectButtonProps}
-                                />
-                            )}
-
-                            {!isEmpty(viewRecordButtonProps) && (
-                                <PersonPanelDetailsActionButton
-                                    // eslint-disable-next-line react/jsx-props-no-spreading
-                                    {...viewRecordButtonProps}
-                                    label="View Record"
-                                    outlined
-                                />
-                            )}
-                        </div>
-                    )}
+                    {
+                        ((!isEmpty(selectButtonProps) && showSelectButton) ||
+                        !isEmpty(viewRecordButtonProps)) && (
```

Comment:
Can we move some of these expressions out to variables calculated before the `return ( ... );`?
e.g.
```JavaScript
const shouldRenderSelectButton = showSelectButton && !isEmpty(selectButtonProps);
const shouldRenderViewButton = !isEmpty(viewRecordButtonProps);

// ...

return (
    // ...
    {(shouldRenderSelectButton || shouldRenderViewButton) && (
        <div
            className={classes.actions}
        >
             {shouldRenderSelectButton && (
                <PersonPanelDetailsActionButton
                    label="Select"
                    // eslint-disable-next-line react/jsx-props-no-spreading
                    {...selectButtonProps}
                />
             )}
             
             {shouldRenderViewButton && (
                <PersonPanelDetailsActionButton
                    // eslint-disable-next-line react/jsx-props-no-spreading
                    {...viewRecordButtonProps}
                    label="View Record"
                    outlined
                />
             )}
        </div>
    )}
);
```

Reply:
Fixed

Reply:
This was resolved satisfactorily in c0008b4.